### PR TITLE
optional cache busting, `post_json` & bearer auth

### PR DESCRIPTION
**Pull request by cristicbz**
Adds three new options for `HTTP` probes:

* `post_json` -- set at the node level, makes a JSON POST request instead of GET.
* `cache_busting_query_parameter` -- set at service level, optionally disables the `?<timestamp>` query parameter that gets added to requests for over-zealous validating API routes.
* `bearer_token_env_var` -- set at service level, expands an environment variable into an `Authorization: Bearer $ENV_VAR` header for the requests. We want to health check certain services that require authentication. We don't want to inject secrets into the config file, but we can do it via an environment variable (this is pretty common in e.g. a Kubernetes scenario).

Here's an example with the new settings:
```toml
[[probe.service]]
id = "api"
label = "API Routes"
cache_busting_query_parameter = false

[[probe.service.node]]
id = "labellers-predict-comments"
label = "Predict Comments"
mode = "poll"

post_json = "{\"uids\":[\"7cdeeb44cd63ae9c.16355\"]}"

replicas = ["https://reinfer.io/api/v1/datasets/dev-health-check/example/labellers/0/predict-comments"]
```

### CODE REVIEW

**Comment by valeriansaliou** [comment]
Thanks for the PR.

I understand the needs for all those 3 additions. However, I would go more generic w/ them, for instance w/ the `post_json` one which could be changed into a generic way to specify which HTTP method to use for this probe (could even be a PATCH/PUT), which raw body to send (if any), and which HTTP headers to send.

This way, a raw probe HTTP request could be forged by a Vigil user. Authentication could be nested into the raw headers property, with a nice side-effect of adding support for both Bearer and Basic auth, as well as any custom authentication schema w/ non-standard HTTP Authentication headers.

To sum up:

1. `cache_busting_query_parameter ` => `http_request_url_bang` (as the cache buster bang used for HTTP probes only)
2. `bearer_token_env_var ` + `post_json ` => `http_request_method` + `http_request_body` + `http_request_headers`
3. Rename existing `http_body_healthy_match` to `http_response_body_healthy_match` (for consistency w/ new parameters)

All new and existing `http_` properties could even be nested in a sub-object named `http` for a clearer configuration.

What do you think?

**Comment by cristicbz** [comment]
Yeah that all makes sense. One outstanding issue is how would handle
environment variable expansion---to avoid storing credentials mixed with
config it would be good if we could expand env vars on headers (and
recently I hit the same issue the slack hook URL). We could make $VAR
expand to env anywhere it appears (with $$ for escaping).

To avoid silently breaking existing configs with dollar signs, we can make
this behaviour disabled by default and enable-able by a global flag---any
suggestion for what section this flag could live in?

**Comment by valeriansaliou** [comment]
I think it's fine expanding it by default w/o the on/off switch. I'll make a "breaking change" release note; Vigil configuration format does not imply much using the dollar sign. Or, we could move this from `$VAR` to `${VAR}`.
